tsdb/chunks/head_chunks_test.go

```diff
@@ -0,0 +1,301 @@
+// Copyright 2020 The Prometheus Authors
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package chunks
+
+import (
+	"encoding/binary"
+	"io/ioutil"
+	"math/rand"
+	"os"
+	"testing"
+	"time"
+
+	"github.com/prometheus/prometheus/tsdb/chunkenc"
+	"github.com/prometheus/prometheus/util/testutil"
+)
+
+func TestHeadReadWriter_WriteChunk(t *testing.T) {
+	hrw, close := testHeadReadWriter(t)
+	defer func() {
+		testutil.Ok(t, hrw.Close())
+		close()
+	}()
+
+	expectedBytes := []byte{}
+	numChunks := 2000
+	nextChunkOffset := uint64(HeadChunkFileHeaderSize + SeriesRefSize + 2*MintMaxtSize)
```

Comment:
So I kept the offset for chunk to be the place where the actual chunk data starts (i.e. the encoding, data length, and the data itself), because, with the chunk reference (the offset), we never try to read the series reference or the mint maxt. But if that looks confusing, I can make the offset to be at the series reference.

Reply:
yes please unless it makes something else more complicated.

Reply:
Done